embed-code-go/test/embedding_instruction_test/embedding_instruction_test.go

```diff
@@ -0,0 +1,480 @@
+	"embed-code/embed-code-go/embedding_instruction"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+type EmbeddingInstructionTestsPreparator struct {
+	rootDir  string
+	testsDir string
+}
+
+func newEmbeddingInstructionTestsPreparator() EmbeddingInstructionTestsPreparator {
+	rootDir, err := filepath.Abs("../../")
+	if err != nil {
+		panic(err)
+	}
+	testsDir, err := filepath.Abs(".")
+	if err != nil {
+		panic(err)
+	}
+	return EmbeddingInstructionTestsPreparator{
+		rootDir:  rootDir,
+		testsDir: testsDir,
+	}
+}
+
+func (testPreparator EmbeddingInstructionTestsPreparator) setup() {
+	os.Chdir(testPreparator.rootDir)
+}
+
+func (testPreparator EmbeddingInstructionTestsPreparator) cleanup() {
+	os.Chdir(testPreparator.testsDir)
+}
+
+func buildConfigWithPreparedFragments() configuration.Configuration {
+	var config = configuration.NewConfiguration()
+	config.DocumentationRoot = "./test/resources/docs"
+	config.CodeRoot = "./test/resources/code"
+	config.FragmentsDir = "./test/resources/prepared-fragments"
+	return config
+}
+
+type buildInstructionParams struct {
+	fragment  string
+	startGlob string
+	endGlob   string
+	closeTag  bool
+}
+
+func buildInstruction(fileName string, params buildInstructionParams) string {
+	fragmentAttr := xmlAttribute("fragment", params.fragment)
+
+	instructionLine := fmt.Sprintf("<embed-code file=\"%s\" %s", fileName, fragmentAttr)
+
+	if len(params.startGlob) > 0 {
+		startAttr := xmlAttribute("start", params.startGlob)
+		instructionLine += " " + startAttr
+	}
+	if len(params.endGlob) > 0 {
+		endAttr := xmlAttribute("end", params.endGlob)
+		instructionLine += " " + endAttr
+	}
+	if params.closeTag {
+		instructionLine += "></embed-code>"
+	} else {
+		instructionLine += "/>"
+	}
+	return instructionLine
+}
+
+func xmlAttribute(name string, value string) string {
+	return fmt.Sprintf("%s=\"%v\"", name, value)
+}
+
+func TestFalseXML(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("FromXML had to raise a panic, but it didn't.")
+		}
+	}()
+
+	xmlString := "<file=\"org/example/Hello.java\" fragment=\"Hello class\"/>"
+	config := buildConfigWithPreparedFragments()
+
+	embedding_instruction.FromXML(xmlString, config)
+}
+
+func TestParseFromXML(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Error: an exception occured on FromXML.")
+		}
+	}()
+
+	instructionParams := buildInstructionParams{}
+	instructionParams.fragment = "Hello class"
+	xmlString := buildInstruction("org/example/Hello.java", instructionParams)
+	config := buildConfigWithPreparedFragments()
+	embedding_instruction.FromXML(xmlString, config)
+}
+
+func TestParseWithClosingTag(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Error: an exception occured on FromXML.")
+		}
+	}()
+
+	instructionParams := buildInstructionParams{}
+	instructionParams.fragment = "Hello class"
+	instructionParams.closeTag = true
```

Comment:
Since test code is pretty complicated I propose to be more verbose and instantiate parameters as an object: 
```
    instructionParams := buildInstructionParams{
        fragment: "Hello class",
        closeTag: true,
    }
```